arangodb/src/main/java/com/yahoo/ycsb/db/ArangoDBClient.java

```diff
@@ -0,0 +1,488 @@
+          logger.error("Failed to create database: " + databaseName, e);
+          System.exit(-1);
+        } else {
+          logger.info("DB already exists: {}", databaseName);
+        }
+      }
+      // Always set the default db
+      arangoDriver.setDefaultDatabase(databaseName);
+      logger.info("ArangoDB client connection created to " + ip + ":" + port);
+      
+      // Log the configuration
+      StringBuilder configStr = new StringBuilder("Arango Configuration: ");
+      configStr.append(String.format("dropDBBeforeRun: %s; ", dropDBBeforeRun));
+      configStr.append(String.format("address: %s:%s; ", ip, port));
+      configStr.append(String.format("databaseName: %s; ", databaseName));
+      configStr.append(String.format("waitForSync: %s; ", waitForSync));
+      configStr.append(String.format("transactionUpdate: %s; ", transactionUpdate));
+      logger.info(configStr.toString());
+    }
+  }
+
+  /**
+   * Can't create automatically during inserting, so create one if necessary.
+   * @param table
+   */
+  private void createCollection(String table) {
+    synchronized(ArangoDBClient.class) {
+      if (defaultCollection == null) {
+        try {
+          CollectionOptions options = new CollectionOptions();
+          options.setWaitForSync(waitForSync);
+          arangoDriver.createCollection(table, options);
+          logger.info("New collection created: {} waitForSync: {}", table, waitForSync);
+        } catch (ArangoException e) {
+          if (e.getErrorNumber() != ErrorNums.ERROR_ARANGO_DUPLICATE_NAME) {
+            logger.error("Fail to create collection", e);
+            System.exit(-1);
+          } else {
+            logger.info("Collection already exists: {}", table);
+          }
+        }
+        defaultCollection = table;
+      }
+    }
+  }
+  
+  /**
+   * Cleanup any state for this DB. Called once per DB instance; there is one
+   * DB instance per client thread.
+   * 
+   * Actually, one client process will share one DB instance here.(Coincide to
+   * mongoDB driver)
+   */
+  @Override
+  public void cleanup() throws DBException {
+    if (INIT_COUNT.decrementAndGet() == 0) {
+      arangoConfigure = null;
+      arangoDriver = null;
+      logger.info("Local cleaned up.");
+    }
+  }
+
+  /**
+   * Insert a record in the database. Any field/value pairs in the specified
+   * values HashMap will be written into the record with the specified record
+   * key.
+   * 
+   * @param table
+   *      The name of the table
+   * @param key
+   *      The record key of the record to insert.
+   * @param values
+   *      A HashMap of field/value pairs to insert in the record
+   * @return Zero on success, a non-zero error code on error. See the
+   *     {@link DB} class's description for a discussion of error codes.
+   */
+  @Override
+  public Status insert(String table, String key, HashMap<String, ByteIterator> values) {
+    try {
+      if (defaultCollection == null) {
```

Comment:
this initial check for null is good to avoid the synchronization check across DB instances, but it's not needed once you make the variable an instance variable. Also, to do this null check properly you'd have to make the variable `volatile` to ensure the JVM properly handled the fact that you're looking at it here outside of the lock.